README.md

```diff
@@ -2,22 +2,81 @@
 
 ## Building
 
+First, ensure that the target system has the necessary build tooling. On Ubuntu, this can be achieved with:
+```
+$ sudo apt install build-essential autogen autoconf libtool cmake tcl shtool
+```
+
+Now, build third-party dependencies and credis itself:
 ```
 git submodule init
 git submodule update
 
 # Install tcmalloc according to
 # https://github.com/gperftools/gperftools/blob/master/INSTALL
 
-pushd redis && env USE_TCMALLOC=yes make -j && popd
-pushd glog && cmake . && make -j install && popd
-pushd leveldb && make -j && popd
+cd redis
```

Comment:
Can you put back the 3 required commands on their 3 lines (redis, glog, leveldb), then in a separate block, mention that build etcd3-cpp/protobuf only if etcd support is desired?   See my comment in src/CMakeLists.txt.

Reply:
I changed them from pushd to cd because pushd is not a POSIX standard utility while cd is. This causes commands to fail when you run them in sh instead of bash.

Reply:
Will change the etcd support stuff in one go!